### PR TITLE
Do unit test and regression test #4

### PR DESCRIPTION
**Pull request by ezrayst**
Supersede #24

### CODE REVIEW

File: tests/gmsh_test.cc [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+//! \brief Check that IO will store right values
+TEST_CASE("GMSH is checked in 3D", "[GMSH][3D]") {
+
+  //! 3D Test of IO Class
+  const unsigned DIM = 3;
+  const double tolerance = 1.E-12;
+
+  auto mesh = std::unique_ptr<Mesh<3, 8>> (new GMSH<3, 8>());
+  mesh->read_mesh("../bin/regression_test.msh");
```
Comment (kks32) [inline tests/gmsh_test.cc]: Could this be passed as an argument? rename `regression_test.msh` to a more meaningful filename of the geometry you are modelling like `cube_test.msh`

**Review by kks32 — CHANGES_REQUESTED** [review]

File: tests/gmsh_test.cc (hunk 1 above)
Comment (ezrayst) [reply tests/gmsh_test.cc]: I can change the name easily... as for the argument, I checked this: https://github.com/philsquared/Catch/blob/master/docs/own-main.md

Could you explain more on how to get this? I can't seem to add `int main` in `regression_test.cc` file

**Comment by ezrayst** [comment]
Have added the ability to parse an argument for regression test

Note that this is the way as it is still under development:
https://github.com/philsquared/Catch/issues/622

**Review by kks32 — CHANGES_REQUESTED** [review]
Could we rebase to develop?

**Review by kks32 — APPROVED** [review]

**Pull request merged**